src/helpers/createUrl.js

```diff
@@ -0,0 +1,21 @@
+function createUrl(hrefOrElement) {
+	let href
+	if (hrefOrElement instanceof Element) {
+		href = hrefOrElement.getAttribute('href') || hrefOrElement.getAttribute('xlink:href');
+	} else {
+		href = hrefOrElement.toString();
+	}
+
+	const url = new URL(href, document.baseURI);
+
+	// Define custom 'address' getter for pathname + query params
+	Object.defineProperty(url, 'address', {
```

Comment:
I don't have any objections against extending the URL object with an `address` getter. This would also work, and I think is a little easier to parse:
```js
class ExtendedURL extends URL {
	get address() {
		return this.pathname + this.search;
	}
}
return new ExtendedURL(href, document.baseURI);
```

Reply:
We could also call it `SwupURL` 🧋